index.js

```diff
@@ -35,58 +39,66 @@ if (!fs.existsSync('./dist')) {
 }
 app.use(express.static('./dist'));
 
-// Map tokens to endpoints
-var registrations = {};
-
 app.post('/register', function(req, res) {
   crypto.randomBytes(32, function(ex, buf) {
     var token = buf.toString('hex');
 
-    registrations[token] = {
+    // save token in database
+    client.hmset(token, {
       endpoint: req.body.endpoint,
       key: req.body.key,
-    };
+    });
+    client.sadd('registrations', token);
 
     res.send(token);
   });
 });
 
 app.post('/unregister', function(req, res) {
-  if (!registrations[req.body.token]) {
-    res.sendStatus(404);
-    return;
-  }
-
-  delete registrations[req.body.token];
-  res.sendStatus(200);
+  var token = req.body.token;
+  client.exists(token, function(err, result) {
+    if (!result) {
+      res.sendStatus(404);
+      return;
+    }
+    client.del(token);
+    client.srem('registrations', token);
```

Comment:
del is there only because we're using the same database as the site (which is wrong)